#3922: Added troubleshooting section to the bottom of the MISO Bareme… (#2777)
miso-lims/miso-lims: docs/admin/baremetal-installation-guide.md

If you encounter other errors migrating the database, make sure that you are using the recommended version of Flyway
(see [Prerequisites](#prerequisites)).

# Server Deployment Troubleshooting

* Whenever unexpected behaviour arises, be sure to check the build logs, located at `${CATALINA_BASE}/logs`. Read through the output of `catalina.out`, `catalina.<date>.log`, and `localhost.<date>.log`.
* Ensure you are using the correct version of Java with Tomcat. This can be checked in `catalina.out`. In case you are not, append/modify the value of `$JAVA_HOME` in `setenv.sh` or `/etc/default/tomcat9`. 
* If file access for `/storage/miso` is causing an error, [this](https://stackoverflow.com/questions/56827735/how-to-allow-tomcat-war-app-to-write-in-folder) may help.

# Development Alternatives

>  If you are don't have the correct version of MySQL installed, this alternative allows the same result without you downloading it. To do this, download [Docker](https://docs.docker.com/get-docker/) to use a Docker container. Creating the container:

Reviewer: ```suggestion
 If you can't or don't want to install the correct version of MySQL, this alternative allows the same result. To do this, download [Docker](https://docs.docker.com/get-docker/) to use a Docker container. Creating the container:
```

>  If you can't or don't want to install the correct version of MySQL installed, this alternative allows the same result without you downloading it. To do this, download [Docker](https://docs.docker.com/get-docker/) to use a Docker container. Creating the container:

Reviewer: ```suggestion
 If you can't or don't want to install the correct version of MySQL, this alternative allows the same result without you downloading it. To do this, download [Docker](https://docs.docker.com/get-docker/) to use a Docker container. Creating the container:
```

>  If using the correct MySQL version for your database server is causing trouble, consider downloading [Docker](https://docs.docker.com/get-docker/) to use a Docker container. Creating the container (note: things prefixed with `VAR_` are a variable, and their value will be specified below):

Reviewer: > If using the correct MySQL version for your database server is causing trouble

This wording sounds like you have the correct version installed and are having some problem with it. This is an alternative to installing the correct MySQL version, whether you're unable or just not wanting to install it.

> things prefixed with `VAR_` are a variable

The convention used in the rest of this document is to represent variables like `$CONTAINER_NAME`

>     docker run --name CONTAINER_NAME -e MYSQL_ROOT_PASSWORD=ROOT_PASSWORD -e MYSQL_DATABASE=DB_NAME -e MYSQL_USER=DB_USERNAME -e MYSQL_PASSWORD=DB_PASSWORD -p 3306:3306 -d mysql:MYSQL_VERSION

Reviewer 2: It's not clear here which are variable names (e.g. `MYSQL_ROOT_PASSWORD`) and which should be variable values (e.g. `ROOT_PASSWORD`). Could you visually distinguish between them?

Where:
* `CONTAINER_NAME` is your desired Docker container name.
* `ROOT_PASSWORD` is the root password to your MySQL.
* `DB_NAME` is the name of the database (e.g. "lims").
* `DB_USERNAME` is the username to access the database (e.g. "tgaclims").
* `DB_PASSWORD` is the password to access the database (e.g. "tgaclims").

> * `MYSQL_VERSION` is the version of MySQL for the database (e.g. "5,7").

Reviewer 2: It might be worth moving this block above the docker command, for clarity.

Author: I think below should be kept for consistency within the document, where variable values are given after they are referenced. Latest commit (commit 4) should give good clarity.

> * `MYSQL_VERSION` is the version of MySQL for the database (e.g. "5,7").

Reviewer 2: For `MYSQL_VERSION` would the version be `5.7` (period instead of comma)?

> * `$MYSQL_VERSION` is the version of MySQL for the database (e.g. "5.7").

Reviewer: Put the version in the command instead of as a variable - you must use the version that MISO is compatible with. You can also specify 8.0 since this will be in the same release as that upgrade

> You do not need to do anything else from **Step 3** if you use this container method.

Reviewer 2: "anything else" makes it sound like there's still something to be done in Step 3. Since it looks like this entirely replaces step 3, what do you think about rewording this to "Skip the database configuration (step 3) if you use this container method." ?
